### ISSUE TITLE
bug: compiled wasm behaves different compares with interpreter

### ISSUE THREAD

**Issue by XinyuShe** [WasmEdge/WasmEdge#3018]
### Summary

The wasmerror (out of bounds memory access) not only does not throw an error, but also executes normally and produces a result.
[filea21.zip](https://github.com/WasmEdge/WasmEdge/files/13275285/filea21.zip)

It's difficult to shrink this randomly generated file (as we don't have the relevant source code) because, as we can see in the below diagram from a.log, line 693 in function 8, which is `i32.load8_u`, is preceded by two calls: `call 7` and `select`. I've tried to simplify the file, but it affects the semantics, causing the issue to disappear.
![image](https://github.com/WasmEdge/WasmEdge/assets/60457190/645155d5-bb40-4376-9125-29702b0b1f3b)
[a.log](https://github.com/WasmEdge/WasmEdge/files/13277350/a.log)

### Current State

Interpreter reports an error
```
[2023-11-10 09:48:48.101] [error] execution failed: out of bounds memory access, Code: 0x88
[2023-11-10 09:48:48.102] [error]     Accessing offset from: 0x100000d54 to: 0x100000d54 , Out of boundary: 0xffffffff
[2023-11-10 09:48:48.102] [error]     In instruction: i32.load8_u (0x2d) , Bytecode offset: 0x00000610
[2023-11-10 09:48:48.102] [error]     When executing function name: "main"
```

AOT produces results
```
-5.74689e-19
0
0
0
0
```

### Expected State

Report "out of bounds memory access"

### Reproduction steps

```bash
wasmedge filea21.wasm main
wasmedge compile filea21.wasm filea21_aot.wasm
wasmedge filea21_aot.wasm main
```

### Components

CLI

### WasmEdge Version or Commit you used

wasmedge version 0.13.4

### Operating system information

Docker version 24.0.5, build ced0996

### Hardware Architecture

Ubuntu 22.04.3 LTS (GNU/Linux 5.15.0-79-generic x86_64)

**Comment by dannypsnl** [comment]
If the issue involves others runtime, please also add their versions and command lines. Your another issue also need to add these, and the file is too huge, tries to shrink it.

**Comment by dannypsnl** [comment]
By the way, have you test the interpreter mode of WasmEdge? That can help contributors reduce the searching scope.

**Comment by dannypsnl** [comment]
You say no different, but it seems give error message as your expected in what you posted.

**Comment by q82419** [comment]
Fixed in the current master (`0.14.0-rc.5`).
